src/analyzer/analyzermanager.cpp

```diff
@@ -0,0 +1,298 @@
+#include "analyzer/analyzermanager.h"
+
+#include <typeinfo>
+#include <QThread>
+
+#include <QtDebug>
+#include <QMutexLocker>
+#include <QListIterator>
+
+#include "library/trackcollection.h"
+#include "mixer/playerinfo.h"
+#include "track/track.h"
+#include "util/compatibility.h"
+#include "util/event.h"
+#include "util/timer.h"
+#include "util/trace.h"
+
+AnalyzerManager* AnalyzerManager::m_pAnalyzerManager = NULL;
+
+//static
+AnalyzerManager& AnalyzerManager::getInstance(UserSettingsPointer pConfig) {
+    if (!m_pAnalyzerManager) {
+        // There exists only one UserSettingsPointer in the app, so it doens't matter
+        // if we only assign it once.
+        m_pAnalyzerManager = new AnalyzerManager(pConfig);
+    }
+    int maxThreads = m_pAnalyzerManager->m_pConfig->getValue<int>(ConfigKey("[Library]", "MaxAnalysisThreads"));
```

Comment:
Actually, the setting was supposed to be a safeguard, and still, there are more safeguards that should control your use case:

Concretely, giving the user the option to control the number of threads allows a way to reduce the usage of resources in case that the machine is not able to keep with the desired audio latency.
Also, the idealthreadcount should work on most places, but its API suggests that it can return -1 if it cannot detect it (i guess that's more for mobile platforms or some OS where the information cannot be retreived).

Finally, on the AnalyzerManager, there is a safety measure where it checks that the idealthreadcount is sane for the machine in use.